compiler/resolution/functionResolution.cpp

```diff
@@ -463,6 +463,35 @@ resolveUninsertedCall(Type* type, CallExpr* call) {
   return call->isResolved();
 }
 
+static void resolveFnForCall(FnSymbol* fn, CallExpr* call)
+{
+  // If 'call' is already on the call stack, do not add it.
+  // If this assertion fail, change it to 'if'.
+  INT_ASSERT(callStack.n == 0 || call != callStack.v[callStack.n-1]);
+
+  // The following is assumed in printCallStack(), which may be invoked
+  // from resolveFns() if we add 'call' to 'callStack'.
+  INT_ASSERT(call->inTree());
+
+  // Push 'call' onto the stack. In case of an error or warning,
+  // this allows the user to see how they got there.
+  callStack.add(call);
+
+  // do real work
+  resolveFns(fn);
+
+  callStack.pop();
+}
+
+void resolveCallAndCallee(CallExpr* call, bool allowUnresolved) {
```

Comment:
The flow here seems unnatural to me.  I think the following states the behavior more clearly

void resolveCallAndCallee(CallExpr* call, bool allowUnresolved) {
  resolveCall(call);

  if (FnSymbol* callee = call->isResolved()) {
    resolveFnForCall(callee, call);
 } else if (allowUnresolved == false) {
   INT_ASSERT(false);
}
}